url89849c8c5b_timmolter_XChart_tgz-pJ8-org_knowm_xchart_standalone_Example1J8/src/org/knowm/xchart/CSVImporter.java

```diff
@@ -0,0 +1,241 @@
+ * chart. Each CSV file in the folder becomes a series on the chart. the CSV file's name becomes the series' name.
+ *
+ * @author timmolter
+ */
+public class CSVImporter {
+
+  public enum DataOrientation {
+
+    Rows, Columns
+  }
+
+  /**
+   * @param path2Directory
+   * @param dataOrientation
+   * @param width
+   * @param height
+   * @param chartTheme
+   * @return
+   */
+  public static XYChart getChartFromCSVDir(String path2Directory, DataOrientation dataOrientation, int width, int height, ChartTheme chartTheme) {
+
+    // 1. get the directory, name chart the dir name
+    XYChart chart = null;
+    if (chartTheme != null) {
+      chart = new XYChart(width, height, chartTheme);
+    }
+    else {
+      chart = new XYChart(width, height);
+    }
+
+    // 2. get all the csv files in the dir
+    File[] csvFiles = getAllFiles(path2Directory, ".*.csv");
+
+    // 3. create a series for each file, naming the series the file name
+    for (int i = 0; i < csvFiles.length; i++) {
+      File csvFile = csvFiles[i];
+      String[] xAndYData = null;
+      if (dataOrientation == DataOrientation.Rows) {
+        xAndYData = getSeriesDataFromCSVRows(csvFile);
+      }
+      else {
+        xAndYData = getSeriesDataFromCSVColumns(csvFile);
+      }
+
+      if (xAndYData[2] == null || xAndYData[2].trim().equalsIgnoreCase("")) {
+        chart.addSeries(csvFile.getName().substring(0, csvFile.getName().indexOf(".csv")), getAxisData(xAndYData[0]), getAxisData(xAndYData[1]));
+      }
+      else {
+        chart.addSeries(csvFile.getName().substring(0, csvFile.getName().indexOf(".csv")), getAxisData(xAndYData[0]), getAxisData(xAndYData[1]), getAxisData(xAndYData[2]));
+      }
+    }
+
+    return chart;
+  }
+
+  /**
+   * @param path2Directory
+   * @param dataOrientation
+   * @param width
+   * @param height
+   * @return
+   */
+  public static XYChart getChartFromCSVDir(String path2Directory, DataOrientation dataOrientation, int width, int height) {
+
+    return getChartFromCSVDir(path2Directory, dataOrientation, width, height, null);
+  }
+
+  /**
+   * Get the series's data from a file
+   *
+   * @param csvFile
+   * @return
+   */
+  private static String[] getSeriesDataFromCSVRows(File csvFile) {
+
+    String[] xAndYData = new String[3];
+
+    BufferedReader bufferedReader = null;
+    try {
+      int counter = 0;
+      String line = null;
+      bufferedReader = new BufferedReader(new FileReader(csvFile));
+      while ((line = bufferedReader.readLine()) != null) {
+        xAndYData[counter++] = line;
+      }
+
+    } catch (Exception e) {
+      System.out.println("Exception while reading csv file: " + e);
+    } finally {
+      if (bufferedReader != null) {
+        try {
+          bufferedReader.close();
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 119:125, 157:163.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.